File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cand =0;
        int cnt =1;
        int n = nums.length;
        for(int i=0;i<n;i++){
            if(cnt==0)cnt++;
            if(cnt==1)cand = nums[i];
            
            if(cand==nums[i])cnt++;
            else cnt--;
        }
        return cand;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cntx=0;
        int x;
        for(auto y:nums){
            if(cntx==0)x=y;
            if(x==y)cntx++;
            else cntx--;
        }
        return x;
    }
};